lib/server.js

```diff
@@ -96,7 +97,22 @@ class Server extends KarmaEventEmitter {
   }
 
   start () {
-    this._injector.invoke(this._start, this)
+    return Promise.all([
+      BundleUtils.bundleResourceIfNotExist('client/main.js', 'static/karma.js'),
+      BundleUtils.bundleResourceIfNotExist('context/main.js', 'static/context.js')
+    ])
+      .then(() => {
+        const config = this.get('config')
+        return NetUtils.getAvailablePort(config.port).then((port) => {
+          config.port = port
+          this._injector.invoke(this._start, this)
+        })
+      })
+      .catch((error) => {
+        this.log.error('Front-end script compile failed with error: ' + error)
```

Comment:
I think using log4js for fatal errors is insufficient: the message gives no stack trace pointing to the problem.
How about adding `console.error('Front-end script compile failed with error: \n', error)` (note comma).

Reply:
Done